### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by indrekj**
AS-3

### CODE REVIEW

**Comment by urmastalimaa** [comment]
+1

File: spec/freddy/consumers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    context 'when logasm logger' do
+      let(:logger) { logasm_class.new }
+      let(:logasm_class) { Class.new }
+
+      before do
+        stub_const('::Logasm', logasm_class)
+      end
+
+      it 'logs the receive event' do
```
Comment (willmore) [inline spec/freddy/consumers_spec.rb]: receive => received

**Comment by indrekj** [comment]
Updated

File: spec/freddy/consumers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    let(:queue_name) { 'salemove' }
+    let(:delivery) do
+      instance_double(Freddy::Delivery,
+        payload: {key: 'value'},
+        correlation_id: 'a1b2'
+      )
+    end
+
+    context 'when logasm logger' do
```
Comment (willmore) [inline spec/freddy/consumers_spec.rb]: `when logasm logger` wording is awkward. `when configured with logasm logger` might be better.

File: spec/freddy/utils_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    context 'when Airbrake is defined' do
+      let(:airbrake) { double }
+
+      before do
+        stub_const('::Airbrake', airbrake)
+      end
+
+      it 'notifies airbrake' do
+        expect(airbrake).to receive(:notify_or_ignore) do |content|
+          expect(content[:error_class]).to eq('name')
```
Comment (willmore) [inline spec/freddy/utils_spec.rb]: The 4 `expect` statements are too verbose for this level. I recommend encapsulating in a helper method or matcher.

File: spec/freddy/utils_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          expect(content[:parameters]).to eq(a: 'b')
+        end
+
+        subject
+      end
+    end
+
+    context 'when Airbrake is not defined' do
+      it 'does nothing' do
+        should eq(nil)
```
Comment (willmore) [inline spec/freddy/utils_spec.rb]: Does this test really do anything?

Comment (indrekj) [inline spec/freddy/utils_spec.rb]: It ensures that the method can be called and no exception is thrown.

File: spec/freddy/utils_spec.rb (hunk 3 above)
Comment (indrekj) [inline spec/freddy/utils_spec.rb]: It is a low level unit test. Shouldn't it be verbose?

Comment (willmore) [inline spec/freddy/utils_spec.rb]: Ideally, a unit test should test one thing only.
Alternative here is to break up the assertions in to different tests for the different qualities of the content.

File: spec/freddy/utils_spec.rb [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        stub_const('::Airbrake', airbrake)
+      end
+
+      it 'notifies airbrake' do
+        expect(airbrake).to receive(:notify_or_ignore).with(
+          error_class: error_class,
+          error_message: error_message,
+          cgi_data: env_attributes,
+          parameters: parameters
+        )
```
Comment (indrekj) [inline spec/freddy/utils_spec.rb]: @willmore what do you think now? It asserts that airbrake is called with correct arguments.

**Comment by stenver** [comment]
+1, though travis fails

**Comment by indrekj** [comment]
Did manual squash. I messed up fixup commits, that's why it failed.

**Pull request merged**